@xen-orchestra/backups/RemoteAdapter.js

```diff
@@ -277,6 +277,11 @@ class RemoteAdapter {
       delta !== undefined && this.deleteDeltaVmBackups(delta),
       full !== undefined && this.deleteFullVmBackups(full),
     ])
+
+    const dirs = new Set(files.map(file => dirname(file)))
+    for(const dir of dirs){
+      await this.cleanVm(dir, {remove: true, onLog: warn })
```

Comment:
IIRC, we discussed about this before: if the merge worker is not disabled, we need to add a task instead.

What's your opinion on this?

Reply:
I wrote a text here before the first review : https://github.com/vatesfr/xen-orchestra/pull/6081#issuecomment-1006732872 , because, as I think I said to you , I'm on the fence for extracted a scheduledCleanVm function

Reply:
Fine by me for now :slightly_smiling_face: